pkg/discovery/network/network.go

```diff
@@ -63,7 +64,7 @@ func (cn *ClusterNetwork) IsComplete() bool {
 	return cn != nil && len(cn.ServiceCIDRs) > 0 && len(cn.PodCIDRs) > 0
 }
 
-func Discover(dynClient dynamic.Interface, clientSet kubernetes.Interface, operatorClient operatorclientset.Interface,
+func Discover(dynClient dynamic.Interface, clientSet kubernetes.Interface, operatorClient client.Client,
```

Comment:
So this introduces yet another client API to callers in addition to the dynamic API and K8s generated API. Perhaps we should just simplify and use the `controller-runtime` API in lieu of the `kubernetes.Interface` as well. I think we still need the dynamic API when the Go type isn't available.